rocks/RocksGrakn.java

```diff
@@ -47,31 +47,35 @@
     private final Path directory;
     private final Options.Database options;
     private final org.rocksdb.Options rocksConfig;
-    private final AtomicBoolean isOpen;
     private final RocksDatabaseManager databaseMgr;
+    private final AtomicBoolean isOpen;
 
-    private RocksGrakn(Path directory, Options.Database options) {
+    protected RocksGrakn(Path directory, Options.Database options, Factory.DatabaseManager databaseMgrFactory) {
         this.directory = directory;
         this.options = options;
+        this.databaseMgr = databaseMgrFactory.databaseManager(this);
         this.rocksConfig = new org.rocksdb.Options()
                 .setCreateIfMissing(true)
                 .setMergeOperator(new UInt64AddOperator());
 
         ExecutorService.init(MAX_THREADS);
-        databaseMgr = new RocksDatabaseManager(this);
         databaseMgr.loadAll();
         isOpen = new AtomicBoolean(true);
     }
 
     public static RocksGrakn open(Path directory) {
-        return open(directory, new Options.Database());
+        return open(directory, new Options.Database(), new RocksFactory());
+    }
+
+    public static RocksGrakn open(Path directory, Factory factory) {
+        return open(directory, new Options.Database(), factory);
     }
 
-    public static RocksGrakn open(Path directory, Options.Database options) {
-        return new RocksGrakn(directory, options);
+    public static RocksGrakn open(Path directory, Options.Database options, Factory factory) {
```

Comment:
`databaseFactory`

Reply:
Will do. It should be `graknFactory` to be precise (look at the type, it is `Factory` not `Factory.Database`)